pull_request_guidelines.md

```diff
@@ -0,0 +1,78 @@
+## Why do we do pull requests?
+
+* Catching and reducing technical debt
+* Catching mistakes
+* Improving development skills
+* Familiarity with the whole codebase
+* Maintaining consistency
+
+## Pull Request Software
+
+* Reviewboard vs. Github
+
+## Creating Pull Requests
+
+* Pull request granularity (how big should my pull requests be?)
+ * Try to keep them under 500 lines.
+* Pull request “single responsibility principle”
+ * Pull requests should have one focus, for example: Adding a new model. Or updating an API.
+ * This will also help keep them short.
+* Pull request description
+ * You should provide bullet points of the changes you are making.
+ * It is an opportunity to offer up any explanation for why you want to make these changes.
+ * Github allows you to use markdown in here, so format it well.
+* When should screenshots be included?
+ * This is usually not necessary in creating a request.
+ * Maybe if the fix you are making is impossible to explain in words?
+* [Git process, feature branches](git_workflow_quick_reference.md)
+* Assigning people vs. posting in slack
+ * When you're on a team, or working with a another specific dev, it's best to directly assign them.
+ * When the PR is more general, post in slack to get someone else's eyes on it.
+
+## Reviewing Pull Requests
+
+* What type of feedback should be given?
+ * Formatting corrections. i.e. spacing and parentheses, spelling errors
+ * Structural suggestions. i.e. order of methods, refactoring ideas
+ * Obvious mistakes. i.e. unused variables or misused APIs
+ * Or just ask clarifying questions if something is confusing.
+ * There are no dumb comments, but there is dumb code.
+* Correctness
+ * There may not be a correct way to do something yet, either because it's not defined in our style guide or multiple ways are acceptable. Go with your gut!
+* When is it necessary to run the code?
+ * It never hurts...
+ * It should be run and tested when there are heavy UI changes or if you are unsure of any particular change.
```

Comment:
Comment on the PR itself and not on the specific commits.  Exception might be if you're reviewing changes made in response to your comments, and they are hard to locate in the PR itself vs. in the specific commit.